### PR TITLE
Add Support for Proxy to connect to HTTPS Servers

### PR DESCRIPTION
**Pull request by lewijacn**
### Description
This change, similar to what has been done in the Traffic-Replayer [here](https://github.com/opensearch-project/opensearch-migrations/commit/7bb1eef8ea1f2a438f609e8aa1c03fb5c57ade2e#diff-b8c531f1b8f2cf3948ca4249f59868e7181dbc734e5a9f531b1ea32ae75f7641), allows the Logging Proxy to connect to a backside HTTPS server by creating an SslContext and allowing insecure connections if specified.

### Issues Resolved
https://opensearch.atlassian.net/browse/MIGRATIONS-1105

### Testing
Docker compose local testing

### Check List
- [ ] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java [hunk 1]
```diff
@@ -73,15 +76,15 @@ static class Parameters {
                 description = "The maximum number of bytes that will be written to a single TrafficStream object.")
         int maximumTrafficStreamSize = 1024*1024;
         @Parameter(required = false,
-                names = {"--destinationHost"},
-                arity = 1,
-                description = "Hostname of the server that the proxy is capturing traffic for.")
-        String backsideHostname = "localhost";
+            names = {"--insecureDestination"},
+            arity = 0,
+            description = "Do not check the backside server's certificate")
```
Comment (gregschohn) [inline TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java:81]: s/backside/destination/.  You're introducing a new term (backside) when you've already defined it as destination.

File: TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java [hunk 2]
```diff
@@ -171,11 +174,46 @@ private static IConnectionCaptureFactory getConnectionCaptureFactory(Parameters
         }
     }
 
+    private static URI convertStringToUri(String uriString) {
```
Comment (gregschohn) [inline TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java:177]: It's probably worth at least calling out in a comment how this is similar to the replayer.

**Review by gregschohn — APPROVED** [review]
Approving this, but I'd like to verify that you've verified that the proxy can make HTTPs connections w/ the docker conpose solution.  The other comments are minor and can be addressed at your own pace.

**Comment by lewijacn** [comment]
> 

This does work with using HTTPS and the docker solution with a couple caveats that may have already been expected. Mainly I wanted to wait until we add something for point 2. before enabling to avoid disruption for others

1.  The elastic search with searchguard Dockerfile needs to be modified to enable searchguard and use `enableTlsConfig.sh` instead of `disableTlsConfig.sh`
2. Auth header needs to be provided in call to proxy and will be passed to destination call. To make this easier for our local docker solution we may should open another task to allow passing in an auth header parameter similar to what we have done for the Traffic Replayer
3. Edit the docker compose file:

Following command for capture proxy:
 ```
/runJavaWithClasspath.sh org.opensearch.migrations.trafficcapture.proxyserver.Main  --kafkaConnection kafka:9092 --destinationUri  https://elasticsearch:9200 --insecureDestination --listenPort 9200 --sslConfigFile /usr/share/elasticsearch/config/proxy_tls.yml
```

File: TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java (hunk 1 above)
Comment (lewijacn) [reply TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/Main.java]: Agreed adjusted

**Pull request merged**